File: servers/assessment/assessments/assessmentDTO/studentAssessment.go
package assessmentDTO

import (
	"github.com/google/uuid"
	"github.com/prompt-edu/prompt/servers/assessment/assessments/assessmentCompletion/assessmentCompletionDTO"
	"github.com/prompt-edu/prompt/servers/assessment/assessments/categoryAssessment/categoryAssessmentDTO"
	"github.com/prompt-edu/prompt/servers/assessment/assessments/scoreLevel/scoreLevelDTO"
	"github.com/prompt-edu/prompt/servers/assessment/evaluations/evaluationDTO"
)

type StudentAssessment struct {
	CourseParticipationID uuid.UUID                                    `json:"courseParticipationID"`
	Assessments           []Assessment                                 `json:"assessments"`
	CategoryAssessments   []categoryAssessmentDTO.CategoryAssessment   `json:"categoryAssessments"`
	AssessmentCompletion  assessmentCompletionDTO.AssessmentCompletion `json:"assessmentCompletion"`
	StudentScore          scoreLevelDTO.StudentScore                   `json:"studentScore"`
	Evaluations           []evaluationDTO.Evaluation                   `json:"evaluations"`
}

File: clients/assessment_component/src/assessment/interfaces/studentAssessment.ts
import { Assessment } from './assessment'
import { AssessmentCompletion } from './assessmentCompletion'
import { CategoryAssessment } from './categoryAssessment'
import { Evaluation } from './evaluation'
import { StudentScore } from './studentScore'

export interface StudentAssessment {
  courseParticipationID: string // UUID
  assessments: Assessment[]
  categoryAssessments: CategoryAssessment[]
  assessmentCompletion: AssessmentCompletion
  studentScore: StudentScore
  evaluations: Evaluation[]
}
